plugins/services/src/js/reducers/form/EnvironmentVariables.js

```diff
@@ -0,0 +1,106 @@
+import {
+  ADD_ITEM,
+  REMOVE_ITEM,
+  SET
+} from '../../../../../../src/js/constants/TransactionTypes';
+import Transaction from '../../../../../../src/js/structs/Transaction';
+
+module.exports = {
+  JSONReducer(state, {type, path, value}) {
+    if (this.env == null) {
+      this.env = [];
+    }
+
+    if (path != null && path.join('.').search('env') !== -1) {
+      if (path.join('.') === 'env') {
+        switch (type) {
+          case ADD_ITEM:
+            this.env.push({key: null, value: null});
+            break;
+          case REMOVE_ITEM:
+            this.env = this.env.filter((item, index) => {
+              return index !== value;
+            });
+            if (this.env.length === 0) {
+              this.env.push({key: null, value: null});
+            }
+            break;
+        }
+
+        return this.env.reduce((memo, item) => {
+          if (item.key != null || item.value != null) {
+            memo[item.key] = item.value;
+          }
+          return memo;
+        }, {});
+      }
+
+      const joinedPath = path.join('.');
```

Comment:
Let' move this up. I see three `path.join('.')`s now.
You could do `const joinedPath = Array.isArray(path) && path.join('.');` (or something similar)